pkg/identityserver/gateway_registry.go

```diff
@@ -398,6 +405,35 @@ func (is *IdentityServer) deleteGateway(ctx context.Context, ids *ttnpb.GatewayI
 	return ttnpb.Empty, nil
 }
 
+func (is *IdentityServer) purgeGateway(ctx context.Context, ids *ttnpb.GatewayIdentifiers) (*types.Empty, error) {
+	if !is.IsAdmin(ctx) {
+		return nil, errAdminsPurgeGateways
+	}
+	err := is.withDatabase(ctx, func(db *gorm.DB) error {
+		// delete related API keys before purging the gateway
+		err := store.GetAPIKeyStore(db).DeleteEntityAPIKeys(ctx, ids)
+		if err != nil {
+			return err
+		}
+		// delete related memberships before purging the gateway
+		err = store.GetMembershipStore(db).DeleteEntityMembers(ctx, ids)
+		if err != nil {
+			return err
+		}
+		// delete related contact info before purging the gateway
+		err = store.GetContactInfoStore(db).DeleteEntityContactInfo(ctx, ids)
+		if err != nil {
+			return err
+		}
+		return store.GetGatewayStore(db).PurgeGateway(ctx, ids)
+	})
+	if err != nil {
+		return nil, err
+	}
+	events.Publish(evtDeleteGateway.NewWithIdentifiersAndData(ctx, ids, nil))
```

Comment:
```suggestion
	events.Publish(evtPurgeGateway.NewWithIdentifiersAndData(ctx, ids, nil))
```

otherwise `gateway.delete` is dispatched

Reply:
I believe the second issue is due to the integrations being stored in AS and its part of a separate issue (https://github.com/TheThingsNetwork/lorawan-stack/issues/2735).